src/qibolab/platforms/abstract.py

```diff
@@ -271,6 +271,12 @@ def transpile(self, circuit: Circuit):
                     mz_pulses.append(MZ_pulse.serial)
                 gate.pulses = tuple(mz_pulses)
 
+            elif isinstance(gate, gates.CZ):
+                cz_sequence = self.create_CZ_pulse_sequence(gate.qubits, sequence.finish)
```

Comment:
Sorry, I missed this before.
These pulses do not need to be added at the end of the sequence, but after the last pulse of the qubit channel.

Reply:
I am sorry but that will not work if one sends a pulse to the passive qubit, so as long as #245 is not sorted I would argue that only `sequence.finish` will work. (passive as in no pulses are sent, as it is not really a control)